transmogrifier/sources/dspace_dim.py

```diff
@@ -0,0 +1,276 @@
+import logging
+from typing import Dict, Iterator, List
+
+from bs4 import Tag
+
+from transmogrifier.models import (
+    AlternateTitle,
+    Contributor,
+    Date,
+    Funder,
+    Identifier,
+    Note,
+    RelatedItem,
+    Rights,
+    Subject,
+    TimdexRecord,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class DSpace_dim:
+    def __init__(
+        self,
+        source: str,
+        source_base_url: str,
+        source_name: str,
+        input_records: Iterator[Tag],
+    ) -> None:
+        self.source = source
+        self.source_base_url = source_base_url
+        self.source_name = source_name
+        self.input_records = input_records
+
+    def __iter__(self) -> Iterator[TimdexRecord]:
+        return self
+
+    def __next__(self) -> TimdexRecord:
+        xml = next(self.input_records)
+        record = self.create_from_dspace_dim(
+            self.source, self.source_base_url, self.source_name, xml
+        )
+        return record
+
+    @classmethod
+    def create_from_dspace_dim(
+        cls, source: str, source_base_url: str, source_name: str, xml: Tag
+    ) -> TimdexRecord:
+        """
+        Args:
+            source: A label for the source repository that is prepended to the
+            timdex_record_id.
+            source_base_url: The base URL for the source system from which direct links
+            to source metadata records can be constructed.
+            source_name: The full human-readable name of the source repository to be
+            used in the TIMDEX record.
+            xml: A BeautifulSoup Tag representing a single DSpace record in
+            dim XML.
+        """
+        # Required fields in TIMDEX
+        source_record_id = cls.create_source_record_id(xml)
+        all_fields = xml.metadata.find_all("field")
```

Comment:
Rather than finding all fields and then searching within those, we can just search directly for the fields we need like we do in other classes, using bs4's option to pass an attribute into the `find_all` method. So titles, for example, would look something like:
```
all_titles= xml.find_all("dim:field", element="title")
```